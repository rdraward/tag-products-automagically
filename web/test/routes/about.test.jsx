import { describe, expect, it, vi } from "vitest";
import renderer from "react-test-renderer";
import About from "../../routes/about";

vi.mock("react-router-dom", async () => {
  const useNavigate = () => vi.fn();
  return { useNavigate };
});

vi.mock("@shopify/polaris", async () => {
  const Page = ({ title, backAction, children }) => {
    return (
      <div>
        <h1>{title}</h1>
        <div>{backAction.content}</div>
        <div>{children}</div>
      </div>
    );
  };
  const Text = ({ variant, as, children }) => {
    return <p>{children}</p>;
  };
  return { Page, Text };
});

// a simple example of a snapshot test
// this is similar to testing basic React components that do not require mocks
describe("snapshot test for frontend/routes/index.jsx", () => {
  it("should render the Index component", () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
