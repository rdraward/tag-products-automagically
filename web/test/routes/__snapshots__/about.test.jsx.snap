// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`snapshot test for frontend/routes/index.jsx > should render the Index component 1`] = `
<div>
  <h1>
    About
  </h1>
  <div>
    Shop Information
  </div>
  <div>
    <p>
      This is a simple Shopify Embedded App.
    </p>
  </div>
</div>
`;
